src/lib/helpers.ts

```diff
@@ -16,3 +17,53 @@ export const blockConfig = (
   startBlock: Math.min(Math.max(start || 0, startBlock), end || Number.MAX_SAFE_INTEGER),
   endBlock: end,
 });
+
+// default rate limit for request per second to RPC endpoints
+const DEFAULT_RPC_RATE_LIMIT = 50;
+
+/**
+ * Reads the RPC request rate limit for a given chain ID from the environment
+ * variable: RPC_REQUEST_RATE_LIMIT_{chainId}.
+ * For example, for Ethereum mainnet the chainId is `1`, so the env variable
+ * can be set as `RPC_REQUEST_RATE_LIMIT_1=400`. This will set the rate limit
+ * for the mainnet (chainId=1) to 400 requests per second. If the environment
+ * variable is not set for the requested chain ID, use `DEFAULT_RPC_RATE_LIMIT`
+ * as the default value.
+ *
+ * The rate limit is the maximum number of requests per second that can be made
+ * to the RPC endpoint. For public RPC endpoints, it is recommended to set
+ * a rate limit to low values (i.e. below 30 rps) to avoid being rate limited.
+ * For private RPC endpoints, the rate limit can be set to higher values,
+ * depending on the capacity of the endpoint. For example, 500 rps.
+ *
+ * @param chainId the chain ID to read the rate limit for from the environment variable
```

Comment:
```suggestion
 * @param chainId the chain ID to get the rate limit for
```

Please see comment above for why suggesting this.